pkg/clients/rds/rds.go

```diff
@@ -124,19 +66,353 @@ func IsErrorNotFound(err error) bool {
+	in.MonitoringRoleARN = awsclients.LateInitializeStringPtr(in.MonitoringRoleARN, db.MonitoringRoleArn)
+	in.MultiAZ = awsclients.LateInitializeBoolPtr(in.MultiAZ, db.MultiAZ)
+	in.PerformanceInsightsKMSKeyID = awsclients.LateInitializeStringPtr(in.PerformanceInsightsKMSKeyID, db.PerformanceInsightsKMSKeyId)
+	in.PerformanceInsightsRetentionPeriod = awsclients.LateInitializeIntPtr(in.PerformanceInsightsRetentionPeriod, db.PerformanceInsightsRetentionPeriod)
+	in.Port = awsclients.LateInitializeIntPtr(in.Port, db.DbInstancePort)
+	in.PreferredBackupWindow = awsclients.LateInitializeStringPtr(in.PreferredBackupWindow, db.PreferredBackupWindow)
+	in.PreferredMaintenanceWindow = awsclients.LateInitializeStringPtr(in.PreferredMaintenanceWindow, db.PreferredMaintenanceWindow)
+	in.PromotionTier = awsclients.LateInitializeIntPtr(in.PromotionTier, db.PromotionTier)
+	in.PubliclyAccessible = awsclients.LateInitializeBoolPtr(in.PubliclyAccessible, db.PubliclyAccessible)
+	in.StorageEncrypted = awsclients.LateInitializeBoolPtr(in.StorageEncrypted, db.StorageEncrypted)
+	in.StorageType = awsclients.LateInitializeStringPtr(in.StorageType, db.StorageType)
+	in.Timezone = awsclients.LateInitializeStringPtr(in.Timezone, db.Timezone)
+
+	if len(in.DBSecurityGroups) == 0 && len(db.DBSecurityGroups) != 0 {
+		in.DBSecurityGroups = make([]string, len(db.DBSecurityGroups))
+		for i, val := range db.DBSecurityGroups {
+			in.DBSecurityGroups[i] = aws.StringValue(val.DBSecurityGroupName)
+		}
+	}
+	if aws.StringValue(in.DBSubnetGroupName) == "" && db.DBSubnetGroup != nil {
+		in.DBSubnetGroupName = db.DBSubnetGroup.DBSubnetGroupName
+	}
+	if len(in.EnableCloudwatchLogsExports) == 0 && len(db.EnabledCloudwatchLogsExports) != 0 {
+		in.EnableCloudwatchLogsExports = db.EnabledCloudwatchLogsExports
+	}
+	if len(in.ProcessorFeatures) == 0 && len(db.ProcessorFeatures) != 0 {
+		in.ProcessorFeatures = make([]v1alpha2.ProcessorFeature, len(db.ProcessorFeatures))
+		for i, val := range db.ProcessorFeatures {
+			in.ProcessorFeatures[i] = v1alpha2.ProcessorFeature{
+				Name:  aws.StringValue(val.Name),
+				Value: aws.StringValue(val.Value),
+			}
+		}
+	}
+	if len(in.VPCSecurityGroupIDs) == 0 && len(db.VpcSecurityGroups) != 0 {
+		in.VPCSecurityGroupIDs = make([]string, len(db.VpcSecurityGroups))
+		for i, val := range db.VpcSecurityGroups {
+			in.VPCSecurityGroupIDs[i] = aws.StringValue(val.VpcSecurityGroupId)
+		}
+	}
+}
+
+func IsUpToDate(in v1alpha2.RDSInstanceParameters, db rds.DBInstance) bool {
+	currentParams := &v1alpha2.RDSInstanceParameters{}
+	LateInitialize(currentParams, db)
+	currentModificationRequest := GenerateModifyDBInstanceInput("", currentParams)
+	candidateModificationRequest := GenerateModifyDBInstanceInput("", &in)
+	return reflect.DeepEqual(currentModificationRequest, candidateModificationRequest)
+}
+
+func GetConnectionDetails(in v1alpha2.RDSInstance) resource.ConnectionDetails {
+	if in.Status.AtProvider.Endpoint.Address == "" {
```

Comment:
Perhaps we could avoid needing to make this check if we setup a secret propagating reconciler, which would ensure any updates to the managed resource secret were propagated to the resource claim secret. There's an example of this pattern in the EKS controller.

https://godoc.org/github.com/crossplaneio/crossplane-runtime/pkg/resource#NewSecretPropagatingReconciler

Reply:
Got it. I thought your intention was to not publish a secret _at all_ until the endpoint is set, but I now see this only contains the endpoint and port.